blenderproc/python/loader/URDFLoader.py

```diff
@@ -0,0 +1,280 @@
+        bpy.ops.object.mode_set(mode='OBJECT')
+
+        # last but not least, we apply constraints
+        if joint_tree.joint_type == "fixed":
+            child.set_location_constraint(x_limits=[0., 0.], y_limits=[0., 0.], z_limits=[0., 0.])
+            child.set_rotation_constraint(x_limits=[0., 0.], y_limits=[0., 0.], z_limits=[0., 0.])
+        elif joint_tree.joint_type == "revolute":
+            if joint_tree.limit is not None:
+                child.set_location_constraint(x_limits=[0., 0.], y_limits=[0., 0.], z_limits=[0., 0.])
+                child.set_rotation_constraint(x_limits=[0., 0.], y_limits=[joint_tree.limit.lower, joint_tree.limit.upper], z_limits=[0., 0.])
+        else:
+            print(f"WARNING: No constraint implemented for joint type '{joint_tree.joint_type}'!")
+
+    # parent links with their children
+    for i, link in enumerate(links):
+        bpy.ops.object.select_all(action='DESELECT')
+        link.select()
+        if i != len(links) - 1:  # select the next link
+            links[i + 1].select()
+
+        # select all visuals, collisions and inertial objects
+        for obj in link.get_children():
+            obj.select()
+
+        # select which object to parent to
+        bpy.context.view_layer.objects.active = link.blender_obj
+
+        # parent object
+        bpy.ops.object.posemode_toggle()
+        bpy.ops.object.parent_set(type="BONE")
+        bpy.ops.object.posemode_toggle()
+
+    # check that the first link is actually the base link, this is just an insanity check
+    assert links[0].get_name() == urdf_tree.base_link.name
+
+    return URDFObject(name=urdf_tree.name, links=links, other_xml=urdf_tree.other_xml)
+
+
+def load_links(link_trees: List["urdfpy.Link"], urdf_path: Union[str, None] = None) -> List[Link]:
+    """ Loads links given a list of urdfpy.Link representations.
+
+    :param link_trees: List of urdf representations of the links.
+    :param urdf_path: Optional path of the urdf file for relative geometry files.
+    :return: List of links.
+    """
+    if not isinstance(link_trees, list):
+        link_trees = list(link_trees)
+    links = []
+    for i, link_tree in enumerate(link_trees):
+        # create armature/bone
+        bpy.ops.object.select_all(action='DESELECT')
+        bpy.ops.object.armature_add()
+        link = Link(bpy_object=bpy.context.active_object)
+
+        # give a name to the link
+        if link_tree.name is not None:
+            link.set_name(name=link_tree.name)
+            print(f"Initialized link {link.get_name()}")
+        else:
+            link.set_name(name=f"link_{i}")
+            print(f"No link name defined for {link_tree}. Set name to link list index: {link.get_name()}")
+
+        # set size
+        if link_tree.visuals or link_tree.collisions:
+            scale = max(get_size_from_geometry(viscol.geometry) for viscol in link_tree.visuals + link_tree.collisions)
```

Comment:
This might create some issues when collisions and visuals have a different scaling

Reply:
That's true. This is taken from Phobos, and I'd argue that in the end it doesn't matter - it only scales the armature which is hidden during rendering anyways. Ideally, what would you like to have here? A check and a warning if size differs?